apollo-core/src/main/java/com/ctrip/framework/apollo/core/utils/DeferredLogCache.java

```diff
@@ -0,0 +1,148 @@
+package com.ctrip.framework.apollo.core.utils;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.slf4j.Logger;
+import org.slf4j.event.Level;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Delayed log printing utility class, used only for logging when Apollo is initialized
+ *
+ * @author kl (http://kailing.pub)
+ * @since 2021/5/11
+ */
+public final class DeferredLogCache {
+
+  public static final int MAX_LOG_SIZE = 1000;
+  private static final AtomicBoolean ENABLED = new AtomicBoolean(false);
```

Comment:
How about we move this ENABLED property to DeferredLogger?

Reply:
It doesn't feel very good, but here you can add a logic that only opens once, what do you think?